metaflow/plugins/kfp/kfp.py

```diff
@@ -1,112 +1,334 @@
 
-    # This is not an ideal way to iterate over the graph, but it's the (easy+)right thing to do for now.
-    # This may need to change as work on improvements.
-    while current_node_name != 'end':
-        for node in graph:
-            if node.name == current_node_name:
-                if current_node_name != 'end':
-                    current_node_name = node.out_funcs[0]
-                    ordered_steps.append(current_node_name)
-                    break
+    # TODO: Metadata needed for client API to run needs to be persisted outside before this
+    print("--------------- RUNNING: CLEANUP OF TEMP DIR----------")
+    subprocess.call(["rm -r /opt/zillow/.metaflow"], shell=True)
 
-    return ordered_steps
+    print("_______________ Done __________________________")
+    return StepOutput(outputs[0], outputs[1])
 
 
-def step_container_op(step_name, code_url=DEFAULT_FLOW_CODE_URL):
+def step_container_op():
     """
-    Method to create a kfp container op that corresponds to a single step in the flow.
+    Container op that corresponds to a step defined in the Metaflow flowgraph.
 
-    TODO: This does not maintain state. The custom pre-start command used below would be removed once we have state accessible across KFP steps.
-    TODO: The public docker is a copy of the internal docker image we were using (borrowed from aip-kfp-example). Check if any stage here may need to be further modified later.
+    Note: The public docker image is a copy of the internal docker image we were using (borrowed from aip-kfp-example).
     """
 
-    python_cmd = """ "python helloworld.py --datastore-root ", $1, " step {} --run-id ", $2, " --task-id ", $4, " --input-paths", $2"/"$5"/"$6 """.format(
-        step_name)
-    command_inside_awk = """ {{ print {0} }}""".format(python_cmd)
-    final_run_cmd = """ python helloworld.py pre-start | awk 'END{}' | sh """.format(command_inside_awk)
+    step_op = kfp.components.func_to_container_op(step_op_func, base_image='ssreejith3/mf_on_kfp:python-curl-git')
+    return step_op
 
-    return dsl.ContainerOp(
 
-        name='StepRunner-{}'.format(step_name),
-        image='ssreejith3/mf_on_kfp:python-curl-git',
-        command= ['sh', '-c'],
-        arguments=[
-            'curl -o helloworld.py {}' \
-            ' && pip install git+https://github.com/zillow/metaflow.git@c722fceffa3011ecab68ce319cff98107cc49532' \
-            ' && export USERNAME="kfp-user" ' \
-            ' && {}'.format(code_url, final_run_cmd)
-            ])
+def initial_setup_container_op():
+    """
+    Container op that corresponds to the 'pre-start' step of Metaflow.
+
+    Note: The public docker image is a copy of the internal docker image we were using (borrowed from aip-kfp-example).
+    """
+
+    init_setup_op = kfp.components.func_to_container_op(initial_setup_op_func, base_image='ssreejith3/mf_on_kfp:python-curl-git')
+    return init_setup_op
+
+
+def add_env_variables(container_op):
```

Comment:
naming: let's call this a transformer function, transformer is a kfp concept.
thanks!

Reply:
Done - changed to `add_env_variables_transformer`